zcash_client_backend/src/proto.rs

```diff
@@ -319,7 +319,6 @@ fn pool_type<T>(pool_id: i32) -> Result<PoolType, ProposalDecodingError<T>> {
     match proposal::ValuePool::try_from(pool_id) {
         Ok(proposal::ValuePool::Transparent) => Ok(PoolType::Transparent),
         Ok(proposal::ValuePool::Sapling) => Ok(PoolType::Shielded(ShieldedProtocol::Sapling)),
-        #[cfg(zcash_unstable = "orchard")]
         Ok(proposal::ValuePool::Orchard) => Ok(PoolType::Shielded(ShieldedProtocol::Orchard)),
         _ => Err(ProposalDecodingError::ValuePoolNotSupported(pool_id)),
```

Comment:
Bug: After removing `#[cfg(zcash_unstable = "orchard")]` here, proposals that contain Orchard spends will be parsed incorrectly. `WalletDb::get_spendable_note` is called with `ShieldedProtocol::Orchard`, but internally it ignores that parameter and always calls `get_spendable_sapling_note`. This may fail with a "note missing" error, but it also may (in the case of an Orchard note received in the same transaction as a Sapling note) return the wrong note. I have not determined what happens after this; I expect something will probably break during transaction creation, but in any case the `Proposal` returned by `zcash_client_backend::proto::proposal::Proposal::try_into_standard_proposal` will be in an invalid state.

Proposals with no Orchard spends, but that request sending to Orchard receivers of UAs, will AFAICT be successfully parsed instead of being rejected with `ProposalDecodingError::ValuePoolNotSupported` (this helper function is called in `zcash_client_backend::proto::proposal::Proposal::try_into_standard_proposal`). I believe what happens instead is that once the user attempts to create transactions from the proposal, the Orchard inputs are ignored, and the resulting transaction fails to balance, which is an unintuitive error.
- EDIT: Nope, currently what happens is that `payment_pools` is being ignored, and the UA's Sapling or transparent receiver would be used, which is a bug. Once that bug is fixed, this will be an unintuitive error.

Reply:
Okay, so this is actually a revealed bug in `zcash_client_sqlite`, not a bug here.